astroscrappy/astroscrappy.pyx

```diff
@@ -51,12 +53,38 @@ def detect_cosmics(indat, inmask=None, float sigclip=4.5, float sigfrac=0.3,
     Parameters
     ----------
     indat : float numpy array
-        Input data array that will be used for cosmic ray detection.
+        Input data array that will be used for cosmic ray detection. This
+        should include the sky background (or a mean background level, added
+        back in after sky subtraction), so that noise can be estimated
+        correctly from the data values.
 
     inmask : boolean numpy array, optional
         Input bad pixel mask. Values of True will be ignored in the cosmic ray
         detection/cleaning process. Default: None.
 
+    bkg : float numpy array, optional
+        A pre-determined background image, to be subtracted from ``indat``
+        before running the main detection algorithm (after deriving any noise
+        estimates). This is used primarily with spectroscopic data, to remove
+        sky lines and the cross-section of an object continuum during
+        iteration, "protecting" them from spurious rejection (see the above
+        paper). This background is not removed from the final, cleaned output
+        (`cleanarr`). The necessary input array can be generated by evaluating
+        a low-order fit to each image row, for both axes, and adding the two
+        resulting images.
+
+    var : float numpy array, optional
+        A pre-determined estimate of the data variance (ie. noise squared) in
+        each pixel, generated by previous processing of ``indat``. If provided,
+        this is used in place of an internal noise model based on ``indat``,
+        ``gain`` and ``readnoise``. This still gets median filtered and cleaned
+        internally, to estimate what the noise in each pixel *would* be in the
+        absence of cosmic rays, but without removing ``bkg`` temporarily (a
+        difference that could lead to less accurate results than the default
+        noise model close to fine structure). This argument should be provided
```

Comment:
Would it just make sense to take the masked median of the variance array? Or just replace the masked regions with the median filter of the var array? This way you wouldn't have to worry about the bkg stuff.